src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -123,7 +124,36 @@ public override bool Execute()
                 if (string.IsNullOrEmpty(request.NuspecFile))
                 {
                     packageBuilder = logic.GetPackageBuilder(request);
+                    string noWarnProperties = null;
+
+                    if (packageBuilder?.Properties?.TryGetValue("NoWarn", out noWarnProperties) == true
+                        && !string.IsNullOrWhiteSpace(noWarnProperties))
+                    {
+                        PackageSpecificWarningProperties packageSpecificWarningProperties = new();
```

Comment:
The `PackageSpecificWarningProperties` class has some methods for creating the object from something else.  I think this code should live there:

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/PackageSpecificWarningProperties.cs